File: util/src/entities/Template.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { Guild } from "./Guild";
import { User } from "./User";

@Entity("templates")
export class Template extends BaseClass {
	@Column({ unique: true })
	code: string;

	@Column()
	name: string;

	@Column({ nullable: true })
	description?: string;

	@Column({ nullable: true })
	usage_count?: number;

	@Column({ nullable: true })
	@RelationId((template: Template) => template.creator)
	creator_id: string;

	@JoinColumn({ name: "creator_id" })
	@ManyToOne(() => User)
	creator: User;

	@Column()
	created_at: Date;

	@Column()
	updated_at: Date;

	@Column({ nullable: true })
	@RelationId((template: Template) => template.source_guild)
	source_guild_id: string;

	@JoinColumn({ name: "source_guild_id" })
	@ManyToOne(() => Guild)
	source_guild: Guild;

	@Column({ type: "simple-json" })
	serialized_source_guild: Guild;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Template.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("templates")]
[Index("Code", Name = "UQ_be38737bf339baf63b1daeffb55", IsUnique = true)]
public partial class Template
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("code", TypeName = "character varying")]
    public string Code { get; set; } = null!;

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("description", TypeName = "character varying")]
    public string? Description { get; set; }

    [Column("usage_count")]
    public int? UsageCount { get; set; }

    [Column("creator_id")]
    public long? CreatorId { get; set; }

    [Column("created_at", TypeName = "timestamp without time zone")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "timestamp without time zone")]
    public DateTime UpdatedAt { get; set; }

    [Column("source_guild_id")]
    public long? SourceGuildId { get; set; }

    [Column("serialized_source_guild", TypeName = "jsonb")]
    public string SerializedSourceGuild { get; set; } = null!;

    [ForeignKey("CreatorId")]
    [InverseProperty("Templates")]
    public virtual User? Creator { get; set; }

    [InverseProperty("Template")]
    public virtual ICollection<Guild> Guilds { get; set; } = new List<Guild>();

    [ForeignKey("SourceGuildId")]
    [InverseProperty("Templates")]
    public virtual Guild? SourceGuild { get; set; }
}
